{{cookiecutter.project_slug}}/server/{{cookiecutter.project_slug}}/core/views.py

```diff
@@ -50,45 +50,45 @@ def post(self, request, *args, **kwargs):
         return Response(response_data)
 
 
-class UserViewSet(
-    viewsets.GenericViewSet,
-    mixins.RetrieveModelMixin,
-    mixins.ListModelMixin,
-    mixins.UpdateModelMixin,
-    mixins.DestroyModelMixin,
-):
+class UserViewSet(viewsets.GenericViewSet, mixins.RetrieveModelMixin):
     queryset = User.objects.all()
     serializer_class = UserSerializer
 
     # No auth required to create user
     # Auth required for all other actions
-    permission_classes = (permissions.IsAuthenticated | CreateOnlyPermissions,)
+    permission_classes = (HasUserPermissions,)
 
-    @transaction.atomic
-    def create(self, request, *args, **kwargs):
+    def get_queryset(self):
         """
-        Endpoint to create/register a new user.
+        Users should only find themselves by default
         """
+        return super().get_queryset().for_user(self.request.user)
```

Comment:
This `get_queryset` doesn't do anything right now. But if you were to add the list mixin or add a `def list()` below then it'll do the job of only returning that logged in user for that list (or all users in a list for staff users)

Reply:
Are you sure? I think get_queryset is also called for update methods